cmd/infra/azure/create.go

```diff
@@ -387,12 +387,13 @@ func buildCreateServicePrincipalCommand(subscriptionID, managedResourceGroupName
 	managedRG := fmt.Sprintf("/subscriptions/%s/resourceGroups/%s", subscriptionID, managedResourceGroupName)
 	nsgRG := fmt.Sprintf("/subscriptions/%s/resourceGroups/%s", subscriptionID, nsgResourceGroupName)
 	vnetRG := fmt.Sprintf("/subscriptions/%s/resourceGroups/%s", subscriptionID, vnetResourceGroupName)
+	dnsZoneRG := fmt.Sprintf("/subscriptions/%s/resourceGroups/os4-common", subscriptionID)
```

Comment:
why is this resource group name hardcoded?

Reply:
Right, that should be a flag. This is just needed for CI/testing purposes. In a real world scenario, the permissions will already be setup prior to the HostedCluster creation.